packages/asset-service/src/generateAssetData/generateAssetData.ts

```diff
@@ -17,6 +19,20 @@ import {
   getZapperTokens
 } from './ethTokens/extendErc20'
 
+const blacklistedAssets: Array<string> = blacklist
+
+export const filterBlacklistedAssets = <T extends BaseAsset | TokenAsset>(assets: T[]) => {
+  const filteredAssets = filter(assets, (token) => {
+    if ('tokens' in token && token.tokens) {
+      token.tokens = filterBlacklistedAssets(token.tokens)
+    }
```

Comment:
A boolean isn't iterable, and the same goes for `undefined`, there would be a runtime error in trying to iterate over false (short-circuit) or `undefined` token.tokens.

Now I get it that this wouldn't happen at runtime but in this case, it means that the types are lying. We shouldn't rely on a `99.9% of the time, it works every time` to write logic, there's always that 0.01%.

The reason why this works is that the `operator in iterable` is making a an implicit discrimination depending on the operator, so why not make it explicit so we can avoid this suboptimal syntax? 

What about:

```suggestion
  const isBaseAsset = (asset: BaseAsset | TokenAsset): asset is BaseAsset =>
    Boolean((asset as BaseAsset)?.tokens)

  const filteredAssets = filter(assets, (token) => {
    if (isBaseAsset(token) && token.tokens) {
      token.tokens = filterBlacklistedAssets(token.tokens)
    }
```